File: bindings/java/src/test/java/io/angzarr/router/conformance/projector/ProjectorFixture.java
package io.angzarr.router.conformance.projector;

import io.angzarr.EventBook;
import io.angzarr.Projection;
import test.counter.Counter;
import test.counter.counter_angzarr;

/** The conformance CounterProjector fixture: every delivered event folds into
 * one projection; the finisher carries the cover and the folded count. */
final class ProjectorFixture implements counter_angzarr.CounterProjectorHandler {

  @Override
  public void increased(Counter.CounterProjectorState.Builder projection, Counter.Increased event) {
    projection.setCount(projection.getCount() + 1);
  }

  @Override
  public Projection finish(Counter.CounterProjectorState.Builder projection, EventBook events) {
    return Projection.newBuilder()
        .setCover(events.getCover())
        .setProjector("counter-projector")
        .setSequence(projection.getCount())
        .build();
  }
}

File: bindings/csharp/conformance/Projector/ProjectorFixture.cs
using Angzarr;
using TC = Test.Counter;

namespace Angzarr.Router.Conformance.Projector;

/// <summary>The conformance CounterProjector fixture: every delivered event
/// folds into one projection; the finisher carries the cover and folded
/// count.</summary>
internal sealed class ProjectorFixture : TC.counter_angzarr.CounterProjectorHandler
{
    public void Increased(TC.CounterProjectorState projection, TC.Increased ev) =>
        projection.Count += 1;

    public Projection Finish(TC.CounterProjectorState projection, EventBook events) =>
        new()
        {
            Cover = events.Cover,
            Projector = "counter-projector",
            Sequence = projection.Count,
        };
}
